contrib/NeonVM/neonvm.c

```diff
@@ -0,0 +1,1346 @@
+    /  We're going to iterate the string table first, to get the count,
+    /  then we'll allocate the data. */
+    const uint8_t *s = start;
+    while (s != end) {
+        s += ((s[0] << 24) | (s[1] << 16) | (s[2] << 8) | s[3]) + 4;
+        (*count)++;
+    }
+
+    r = malloc((sizeof(uint8_t *)) * *count);
+    if (r == NULL) {
+        fatal_error("Could not allocate memory for %d strings.", *count);
+    }
+    pBytecode->strings = malloc(sizeof(TString *) * *count);
+
+    while (start != end) {
+        TString *ts = malloc(sizeof(TString));
+        size_t len = (start[0] << 24) | (start[1] << 16) | (start[2] << 8) | start[3];
+        start += 4;
+
+        ts->length = len;
+        ts->data = malloc(len+1); /* Always add null termination regardless of length. */
+        //r[i] = malloc(len+1);
+        //if (r[i] == NULL) {
+        //    fatal_error("Could not allocate %d bytes for string index %d in string table.", len + 1, i);
+        //}
+        //memcpy(r[i], start, len);
+        memcpy(ts->data, start, len);
+        //r[i][len] = '\0';
+        ts->data[len] = '\0'; /* Null terminate all strings, regardless of string type. */
+        pBytecode->strings[i++] = ts;
+        start += len;
+    }
+    return r;
+}
+
+struct tagTExecutor;
+
+void exec_loop(struct tagTExecutor *self);
+int exec_run(struct tagTExecutor *self);
+
+TBytecode *bytecode_newBytecode()
+{
+    TBytecode *pBytecode = malloc(sizeof(TBytecode));
+    if (!pBytecode) {
+        fatal_error("Could not allocate memory for neon bytecode.");
+    }
+
+    memset(pBytecode, 0x00, sizeof(TBytecode));
+    return pBytecode;
+}
+
+void bytecode_freeBytecode(TBytecode *b)
+{
+    uint32_t i;
+    if (!b) {
+        return;
+    }
+
+    free(b->pFunctions);
+    free(b->imports);
+    free(b->export_functions);
+    free(b->pExportTypes);
+    for (i = 0; i < b->strtablelen; i++) {
+        free(b->strings[i]->data);
+        free(b->strings[i]);
+    }
+    free(b->strtable);
+    free(b->strings);
+    free(b);
+    b = NULL;
+}
+
+typedef struct tagTExecutor {
+    struct tagTBytecode *object;
+    uint32_t ip;
+    struct tagTStack *stack;
+    /*struct tagTStack *callstack; */
+    uint32_t callstack[300];
+    int32_t callstacktop;
+    int32_t param_recursion_limit;
+    Cell *globals;
+    TFrameStack *framestack;
+    //uint64_t frames;
+
+    /* Debug / Diagnostic fields */
+    uint64_t total_opcodes;
+    uint64_t callstack_depth;
+    clock_t time_start;
+    clock_t time_end;
+} TExecutor;
+
+void exec_freeExecutor(TExecutor *e)
+{
+    uint32_t i;
+
+
+
+    for (i = 0; i < e->object->global_size; i++) {
+        cell_clearCell(&e->globals[i]);
+    }
+    free(e->globals);
+    destroyStack(e->stack);
+    free(e->stack);
+    destroyFrameStack(e->framestack);
+    free(e->framestack);
+
+    bytecode_freeBytecode(e->object);
+
+    free(e);
+}
+
+struct tagTExecutor *exec_newExecuter(struct tagTBytecode *object);
```

Comment:
Spelling of "Executer" doesn't match "Executor" everywhere else.

Reply:
Removed all references to Executer.